java-checks/src/main/java/org/sonar/java/checks/AnonymousClassShouldBeLambdaCheck.java

```diff
@@ -72,48 +75,49 @@ public void visitNewClass(NewClassTree tree) {
     if (classBody != null) {
       TypeTree identifier = tree.identifier();
       if (!useThisIdentifier(classBody) && !enumConstants.contains(identifier) && isSAM(classBody)) {
-        context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        Tree member = classBody.members().get(0);
+        if (member.is(Tree.Kind.METHOD) && multipleMethodsDefaultSameSignature(((JavaSymbol.MethodJavaSymbol) ((MethodTree) member).symbol()))) {
+          context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        }
       }
     }
   }
 
   private static boolean isSAM(ClassTree classBody) {
     if (hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
       // Verify class body is a subtype of an interface
-      MethodJavaSymbol methodSymbol = ((MethodJavaSymbol) ((MethodTree) classBody.symbol().memberSymbols().stream().findFirst().get().declaration())
-        .symbol());
       JavaSymbol.TypeJavaSymbol symbol = (JavaSymbol.TypeJavaSymbol) classBody.symbol();
-      boolean checkAnonymousClassFP = defaultMehodsSameSign(methodSymbol);
       return symbol.getInterfaces().size() == 1 &&
-        symbol.getSuperclass().is("java.lang.Object") &&
-        checkAnonymousClassFP;
+        symbol.getSuperclass().is("java.lang.Object");
     }
     return false;
   }
 
-  // seaches in the interface if there is a default method with the same signature
-  private static boolean defaultMehodsSameSign(MethodJavaSymbol methodSymbol) {
-    TypeJavaSymbol tjs = methodSymbol.overriddenSymbol().outermostClass();
-    if (methodSymbol.overriddenSymbol().isDefault() && tjs.isInterface() && tjs.declaration() != null) {
-      for (Symbol mjs : ((TypeJavaSymbol) tjs.declaration().symbol()).memberSymbols()) {
-        if (mjs.isMethodSymbol() && !checkmethodSignature(methodSymbol, (MethodJavaSymbol) mjs)) {
-          return false;
-        }
-      }
+  private static boolean multipleMethodsDefaultSameSignature(JavaSymbol.MethodJavaSymbol methodSymbol) {
+    JavaSymbol.TypeJavaSymbol typeJavaSymbol = methodSymbol.overriddenSymbol().enclosingClass();
```

Comment:
I think your solution won't work because of that.

Because you are starting from the owner of the overriden symbol, you are missing all the other cases which could be below in the hierarchy. See my example in other comment.